### PR TITLE
Bench 105 create docker image

### PR DESCRIPTION
**Pull request by PietroConvalleAD**
Adds DockerFile and docker-compose.yml

### CODE REVIEW

**Review by MichaelOSullivanAnswer — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -19,13 +19,13 @@ drf-problems = { git = "https://github.com/Axeltherabbit/drf-problems" }
...
 tzdata = "^2022.7"
+coverage = "^6.5.0"
+drf-yasg = "^1.21.4"
 
 [tool.poetry.group.dev.dependencies]
 black = "^22.10.0"
 pyright = "^1.1.278"
-coverage = "^6.5.0"
 pycodestyle = "^2.9.1"
-drf-yasg = "^1.21.4"
```
Comment (PietroConvalleAD) [inline pyproject.toml:26]: why were these moved to main dependencies?

File: pyproject.toml [hunk 2]
```diff
@@ -17,13 +17,15 @@ django-json-404-middleware = { git = "https://github.com/Axeltherabbit/django-js
...
+tzdata = "^2022.7"
+coverage = "^6.5.0"
+drf-yasg = "^1.21.4"
 
 [tool.poetry.group.dev.dependencies]
 black = "^22.10.0"
 pyright = "^1.1.278"
-coverage = "^6.5.0"
 pycodestyle = "^2.9.1"
-drf-yasg = "^1.21.4"
```
Comment (PietroConvalleAD) [inline pyproject.toml]: why were these moved to main dependencies?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+RUN pip install poetry
+COPY . /AnswerKing-Python
+WORKDIR /AnswerKing-Python
+#ENV PATH="${PATH}:/root/.local/bin"
+RUN poetry install --only main
+
+CMD [ "/usr/local/bin/poetry", "run", "python", "manage.py", "migrate", "0.0.0.0:8000"]
+CMD [ "/usr/local/bin/poetry", "run", "python", "manage.py", "runserver", "0.0.0.0:8000"]
+EXPOSE 8000
```
Comment (PietroConvalleAD) [inline Dockerfile]: Expose is defined in docker-compose

File: pyproject.toml (hunk 2 above)
Comment (JoeCSykes) [reply pyproject.toml]: because it was erroring saying the coverage module and the drf-yasg modules were not installed. They weren't being installed because they were in the dev dependencies and you were running poetry install --only-main. As such I moved them into main so it didn't error

Comment (PietroConvalleAD) [reply pyproject.toml]: of course, you have to set your .env to use the .base settings.py.

Comment (JoeCSykes) [reply pyproject.toml]: I don't understand. this seems a lot more complicated than just including them in the main dependencies

Comment (PietroConvalleAD) [reply pyproject.toml]: uh? no, it's what DJANGO_SETTINGS_MODULE= in the .env file was made for

Comment (PietroConvalleAD) [reply pyproject.toml]: `DJANGO_SETTINGS_MODULE=answerking.settings.base`

File: .env.production [hunk 4]
```diff
@@ -0,0 +1,8 @@
+DATABASE_NAME=answerking_app
+DATABASE_HOST=host.docker.internal
+DATABASE_PORT=3306
+DATABASE_USER=root
+DATABASE_PASS=Admin123
+SECRET_KEY="django-insecure-x977=v5a2q-e%_5$b9ge@jzk%s_nhk2l8_use&h@*m$w33dopr"
+DJANGO_SETTINGS_MODULE=answerking.settings.base
+DATABASE_ENGINE = "django.db.backends.mysql"
```
Comment (PietroConvalleAD) [inline .env.production]: why? use github secrets as you'd normally do, you might have to override it in the workflow using another variable `DJANGO_SETTINGS_MODULE = {{GITHUB.DJANGO_SETTINGS_MODULE_DEPLOY}}`

**Comment by PietroConvalleAD** [comment]
Does build now work if you switch the merge to release?

**Comment by JoeCSykes** [comment]
no it should run when someone pushes to a branch named release-*

File: .env.production (hunk 4 above)
Comment (JoeCSykes) [reply .env.production]: this si when you are running it locally so it wouldn't have access to github secrets

**Comment by PietroConvalleAD** [comment]
Yeah I mean, just switch it to release-test to test it maybe

**Comment by JoeCSykes** [comment]
Created a release-test branch and the docker-image was built and deployed to git hub. This can be seen in the packages section of git hub which is tagged with release-test

**Comment by PietroConvalleAD** [comment]
nice, good job

File: README.md [hunk 5]
```diff
@@ -43,3 +45,24 @@ Commands for maintaining consistency and PEP8 standards across codebase, as well
...
+- create a .env.production file containing (filling out the database password with your chosen password):
+```
+DATABASE_NAME=answerking_app
+DATABASE_HOST=host.docker.internal
+DATABASE_PORT=3306
+DATABASE_USER=root
+DATABASE_PASS=
+SECRET_KEY="django-insecure-x977=v5a2q-e%_5$b9ge@jzk%s_nhk2l8_use&h@*m$w33dopr"
+DJANGO_SETTINGS_MODULE=answerking.settings.base
+DATABASE_ENGINE = "django.db.backends.mysql"
```
Comment (NogaAD) [inline README.md]: Remove spaces around '='

Comment (JoeCSykes) [reply README.md]: done

**Review by NogaAD — APPROVED** [review]
